File: BestTimeToBuyAndSellCrypto/best_time_to_buy_and_sell_crypto.py
#!/usr/bin/env python3

class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        max_prof = 0
        buy_price = prices[0]

        for price in prices[1:]:
            if price > buy_price:
                max_prof = max(price - buy_price, max_prof)
            else:
                buy_price = price
        return max_prof

File: BestTimeToBuyAndSellCrypto/best_time_to_buy_and_sell_crypto.cpp
#include <algorithm>
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int buy_price = 101, profit = 0;
        for (int price : prices){
            if (price < buy_price){
                buy_price = price;
            }
            else{
                profit = std::max(profit, price - buy_price);
            }
        }
        return profit;
    }
};

